### DISCUSSION TITLE
Werf and Hierarchical Namespace Controller (HNC)

### DISCUSSION THREAD

**Discussion by slavb18** [werf/werf#5923] [General]
Can I configure Werf to use [Hierarchical Namespace Controller (HNC)](https://github.com/kubernetes-sigs/hierarchical-namespaces) ?

E.g. company configures parent namespace `mycompany`, and werf creates child namespaces using `kubectl hns create mycompany-myapp-production -n mycompany`

Or should I just create required namespace in script before running werf?

**Comment by ilya-lesikov** [comment]
Doesn't it work now? I'm not quite sure, but feels like it should work as your describe. Creating namespace(s) manually before deployment is also fine.

**Reply by slavb18** [reply]
Werf creates plain namespaces (like using `kubectl create namespace` or by manifests)
Don't know how to configure werf to  use hns: `kubectl hns create mycompany-myapp-production -n mycompany`

**Reply by ilya-lesikov** [reply]
Well, I see. There is no support for creation of such namespaces from inside werf. Are there plans to make HNS a part of the Kubernetes core? If so, we'll need to add support for this, otherwise not sure.

Anyways, automatic namespace creation from inside werf is more of a shortcut at this time. If it's not enough, use `kubectl create ns` or `kubectl create hns`.

We'll consider adding support for HNS though.
